### PR TITLE
Add support for TopologySpreadConstraints & nodeSelector on collector and target allocator

### LINKED ISSUE #1899: otelcollector CRD missing TopologySpreadConstraints and NodeSelector
Would be nice to be able to specify TopologySpreadConstraint for otel-collector and TA in opentelemetry-collector CRD. There is also no nodeSelector possible on TA right now.

I'd be happy to file a PR for that.

### PR DESCRIPTION
**Pull request by cmergenthaler**
Add support for TopologySpreadConstraints on collector and target allocator. Also adding nodeSelector for TA.

Fixes #1899

### CODE REVIEW

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 1]
```diff
@@ -199,6 +205,10 @@ type OpenTelemetryTargetAllocator struct {
 	// that can be run in a high availability mode is consistent-hashing.
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
+	// NodeSelector to schedule OpenTelemetry Collector pods.
```
Comment (jaronoff97) [inline apis/v1alpha1/opentelemetrycollector_types.go:208]: This should say TA pods here

File: pkg/targetallocator/deployment_test.go [hunk 2]
```diff
@@ -17,13 +17,28 @@ package targetallocator
 import (
 	"testing"
 
+	v1 "k8s.io/api/core/v1"
```
Comment (jaronoff97) [inline pkg/targetallocator/deployment_test.go:20]: I think this import should be in the group below

**Review by jaronoff97 — COMMENTED** [review]
Just a few small things, thanks for adding this!

File: apis/v1alpha1/opentelemetrycollector_types.go (hunk 1 above)
Comment (cmergenthaler) [reply apis/v1alpha1/opentelemetrycollector_types.go]: Done

File: pkg/targetallocator/deployment_test.go (hunk 2 above)
Comment (cmergenthaler) [reply pkg/targetallocator/deployment_test.go]: Done

**Review by frzifus — APPROVED** [review]

**Review by iblancasa — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
Thank you for your contribution! 🎉🎉🎉

**Pull request merged**